rollup/main/default/classes/Rollup.cls

```diff
@@ -1631,13 +1645,51 @@ global without sharing virtual class Rollup implements Database.Batchable<SObjec
       }
       String actualFieldName = entityDefinitionToFields.get(currentObject).get(desiredField);
       return fieldNameToField.get(actualFieldName);
-    } catch (Exception ex) {
-      // do nothing, it didn't work
     }
 
     return null;
   }
 
+  private static Map<String, DescribeSObjectResult> durableIdToDescribeToken = new Map<String, DescribeSObjectResult>();
```

Comment:
Would it be better to go back to using a SOQL query for `Rollup__mdt` at this point, instead of `Rollup__mdt.getAll()`? You're doing some additional queries + code to try to determine the string names for custom object/fields, but you can get it for both standard & custom objects/fields with something like this:

```
List<Rollup__mdt> rollups = [
    select id, LookupObject__r.QualifiedApiName, CalcItem__r.QualifiedApiName, LookupFieldOnCalcItem__r.QualifiedApiName, RollupFieldOnLookupObject__r.QualifiedApiName 
    from Rollup__mdt
];
for (Rollup__mdt rollup : rollups) {
    System.debug('rollup.LookupObject__r.QualifiedApiName==' + rollup.LookupObject__r.QualifiedApiName);
    System.debug('rollup.CalcItem__r.QualifiedApiName==' + rollup.CalcItem__r.QualifiedApiName);
    System.debug('rollup.RollupFieldOnLookupObject__r.QualifiedApiName==' + rollup.RollupFieldOnLookupObject__r.QualifiedApiName);
    System.debug('rollup.LookupFieldOnCalcItem__r.QualifiedApiName==' + rollup.LookupFieldOnCalcItem__r.QualifiedApiName);
}
```

![image](https://user-images.githubusercontent.com/1267157/109919294-4a043000-7c6d-11eb-9e3f-3cefe885a801.png)

Reply:
Yeah, I totally agree. I really wanted to love the `getAll()` CMDT methods exposed in Spring 21, but they don't return the parent fields and the hoops I was jumping through to try to "solve" the CMDT field definition/entity definition issues were compounding. Now that I've standardized _where_ the `Rollup__mdt` records come from, going back to the query is an excellent call - it removed > 100 lines of code purely related to trying to massage the values to something sensible!